File: app/src/main/java/com/example/galaxyplayerkotlin/Activities/MainActivityClass.java
package com.example.galaxyplayerkotlin.Activities;


import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.example.galaxyplayerkotlin.Adapters.SimpleSectionPagerAdapter;
import com.example.galaxyplayerkotlin.Fragments.LoginFragment;
import com.example.galaxyplayerkotlin.Fragments.MusicPlayerFragment;
import com.example.galaxyplayerkotlin.Fragments.SongListFragment;
import com.example.galaxyplayerkotlin.Fragments.SwapiFragment;
import com.example.galaxyplayerkotlin.R;
import com.google.android.material.tabs.TabLayout;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

public class MainActivityClass extends AppCompatActivity {

    static ViewPager viewPager;
    public SimpleSectionPagerAdapter simpleSectionPagerAdapter;
    TabLayout tabLayout;

    private static final String TAG = "MainActivityClass";


    public static boolean permissionAllowed;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_layout);

        tabLayout = findViewById(R.id.tabLayout_id);

        viewPager = findViewById(R.id.view_pager);

        simpleSectionPagerAdapter = new SimpleSectionPagerAdapter(getSupportFragmentManager());

        //sectionPageAdapter = new SectionPagerAdapter(getSupportFragmentManager());

        viewPager.setAdapter(simpleSectionPagerAdapter);

        Log.d(TAG, "onCreate: sectionPageAdapter set to viewPager");

        simpleSectionPagerAdapter.addFragment(new MusicPlayerFragment(), "Music Player");
        simpleSectionPagerAdapter.addFragment(new SwapiFragment(), "Swapi");

        simpleSectionPagerAdapter.notifyDataSetChanged();

        tabLayout.setupWithViewPager(viewPager);

        Log.d(TAG, "onCreate: Opening Login Fragment");
    }


}

File: app/src/main/java/com/example/galaxyplayerkotlin/Activities/MainActivityClass.kt
package com.example.galaxyplayerkotlin.Activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import androidx.viewpager2.widget.ViewPager2
import com.example.galaxyplayerkotlin.Adapters.ViewPagerAdapter
import com.example.galaxyplayerkotlin.R
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator

class MainActivityClass : AppCompatActivity(R.layout.activity_main_layout) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        window
            .statusBarColor = ContextCompat
            .getColor(this, R.color.statusBarColor)


        val viewPager: ViewPager2 = findViewById(R.id.viewPager)
        val tabLayout: TabLayout = findViewById(R.id.tabLayout)

        val adapter = ViewPagerAdapter(supportFragmentManager, lifecycle)
        viewPager.adapter = adapter

        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
            tab.text = fragmentNames[position]
        }.attach()

        viewPager.reduceDragSensitivity()



    }

    private fun ViewPager2.reduceDragSensitivity() {
        val recyclerViewField = ViewPager2::class.java.getDeclaredField("mRecyclerView")
        recyclerViewField.isAccessible = true
        val recyclerView = recyclerViewField.get(this) as RecyclerView

        val touchSlopField = RecyclerView::class.java.getDeclaredField("mTouchSlop")
        touchSlopField.isAccessible = true
        val touchSlop = touchSlopField.get(recyclerView) as Int
        touchSlopField.set(recyclerView, touchSlop*5)       // "8" was obtained experimentally
    }




    val fragmentNames = arrayOf(
        "Tracks",
        "Artists"
    )
}